lib/SIL/IR/SILGlobalVariable.cpp

```diff
@@ -109,6 +109,8 @@ bool SILGlobalVariable::isValidStaticInitializerInst(const SILInstruction *I,
     case SILInstructionKind::BuiltinInst: {
       auto *bi = cast<BuiltinInst>(I);
       switch (M.getBuiltinInfo(bi->getName()).ID) {
+        case BuiltinValueKind::ZeroInitializer:
```

Comment:
Can you add a check here, which returns false if it's an arbitrary width integer?